locust/scenarios/locustfile.py

```diff
@@ -0,0 +1,88 @@
+import logging
+import os
+import subprocess
+import time
+
+from locust import HttpUser, task, between
+from urllib.parse import urlparse
+
+
+url = str(os.environ.get("HOST"))
+parsed_url = urlparse(url)
+host = parsed_url.hostname
+port = parsed_url.port
+
+logger_level = str(os.environ.get("LOGGER_LEVEL"))
+logger = logging.getLogger(__name__)
+logger.setLevel(logger_level)
+
+curve = str(os.environ.get("CURVE"))
+
+class QscNginxUser(HttpUser):
+    wait_time = between(1, 2)
+
+    def on_start(self):
+        self.client.base_url = host
+        logger.info("Starting Locust test using OpenSSL for TLS connection")
+
+    def make_post_quantum_request_with_openssl(self, endpoint):
+        try:
+            logger.debug(f"Making request to {url}{endpoint} with curve {curve}")
+            host_and_port = host + ":" + str(port)
+            http_headers="Post-quantum"
+            request_name = f"Curve {curve} {endpoint}"
+
+            start = time.time()
+            result = subprocess.run(
+                ["openssl", "s_client", "-curves", curve, "-connect", host_and_port, "-ign_eof"],
+                input=f"GET {endpoint} HTTP/1.1\r\n"
+                      f"Host: {host}\r\n"
+                      f"User-Agent: {http_headers}\r\n"
+                      f"Connection: close\r\n\r\n",
+                capture_output=True, text=True
+            )
```

Comment:
As I comment briefly in the README, I decided to use subprocess with openssl s_client in this implementation to handle post-quantum cryptographic curves. The reason for this approach is that current Python libraries, such as urllib and gevent, do not yet support these cryptographic standards.

By invoking openssl through subprocess, we can leverage its built-in support for post-quantum curves and ensure secure communication during our performance tests. This allows us to test the behavior of our endpoints with these  security protocols, without waiting for native support in Python libraries.

I also performed a test by recompiling Python with the OQS (Open Quantum Safe) version of OpenSSL to enable the necessary post-quantum curves. However, even as Python libraries evolve to support these curves, they will still rely on an underlying version of OpenSSL that supports post-quantum cryptography. So, even when native Python support is available, it will still be essential to ensure that the system's OpenSSL version has the required cryptographic capabilities.

Reply:
Worthwhile taking a look at (maybe eventually using) https://github.com/open-quantum-safe/liboqs-python/pull/94?

Reply:
Yes, I have looked into it, but after reviewing it, I believe this implementation requires a thorough investigation to assess whether this wrapper meets our performance standards (it's essential for any performance tool, to exhibit excellent performance behavior). For this initial version, I’d prefer not to include a wrapper without this research. Perhaps we can revisit this in future releases?